spec/forms/provider_interface/interview_wizard_spec.rb

```diff
@@ -0,0 +1,157 @@
+          Timecop.freeze(2021, 1, 13) do
+            expect(wizard).to be_invalid
+            expect(wizard.errors[:date]).to include 'Enter a date that is in the future'
+          end
+        end
+      end
+
+      context 'when it is after the rdb date' do
+        let(:application_choice) { build_stubbed(:application_choice, reject_by_default_at: Time.zone.local(2021, 2, 14)) }
+        let(:day) { 15 }
+
+        it 'is invalid with the correct error' do
+          Timecop.freeze(2021, 1, 13) do
+            expect(wizard).to be_invalid
+            expect(wizard.errors[:date]).to include 'The interview date must be before the application closing date'
+          end
+        end
+      end
+    end
+
+    describe '#time_is_valid?' do
+      let(:tomorrow) { 1.day.from_now }
+      let(:day) { tomorrow.day }
+      let(:month) { tomorrow.month }
+      let(:year) { tomorrow.year }
+
+      context 'checks if the time is in the rights format' do
+        let(:invalid_times) { %w[noon 1700 12:30 12:3pm 1800pm] }
+        let(:valid_times) { %w[12:30pm 2pm 1.30am 01.24AM 3\ 30am] }
+
+        it 'returns false when the time is invalid' do
+          invalid_times.each do |time|
+            wizard.time = time
+            expect(wizard.time_is_valid?).to eq(false)
+          end
+        end
+
+        it 'returns true when the time is valid' do
+          valid_times.each do |time|
+            wizard.time = time
+            expect(wizard.time_is_valid?).to eq(true)
+          end
+        end
+      end
+    end
+  end
+
+  describe '#date_and_time' do
+    let(:time) { '4:30pm' }
+    let(:day) { '20' }
+    let(:month) { '2' }
+    let(:year) { '2022' }
+
+    it 'converts the :date and :time to valid datetime' do
+      expect(wizard.date_and_time.hour).to equal(16)
+      expect(wizard.date_and_time.min).to equal(30)
+      expect(wizard.date_and_time.day).to equal(20)
+      expect(wizard.date_and_time.month).to equal(2)
+      expect(wizard.date_and_time.year).to equal(2022)
+    end
+  end
+
+  describe 'Wizard ' do
+    let(:state_store) { WizardStateStores::SessionStore.new(session: { 'key' => {}.to_json }, key: 'key') }
+
+    describe '#next_step' do
+      it 'returns input' do
+        wizard = described_class.new(state_store, current_step: 'input')
+        expect(wizard.next_step).to eq('check')
+      end
+    end
+
+    describe '#previous_step' do
+      it 'returns new' do
+        wizard = described_class.new(state_store, current_step: 'check')
+        expect(wizard.previous_step).to eq('input')
+      end
+    end
+  end
+
+  describe '#providers_that_user_has_make_decisions_for' do
+    let(:application_choice) { create(:application_choice, :awaiting_provider_decision, course_option: course_option) }
+    let(:course_option) { create(:course_option, course: course) }
+    let(:provider) { create(:provider) }
+
+    context 'when the user has make_decision permissions on the application choice provider only' do
```

Comment:
It's a bit clearer to refer to training provider vs accredited provider for a course
```suggestion
    context 'when the user has make_decision permissions on the application choice training provider only' do
```